src/navigation/navigation.cc

```diff
@@ -325,40 +341,47 @@ void Navigation::predictCollisions(PathOption& path){
 	path.free_path_length = fpl_min;
 }
 
+// New base_link frame clearance calc
 void Navigation::calculateClearance(PathOption &path){
-
-	float turning_radius = 1/path.curvature;
-	float abs_radius = abs(turning_radius);
-
-	// Determine if the car is turning left or right (the math is slightly different for each)
-	std::string turning_direction = (path.curvature < 0 ? "right" : "left");
-
-	// Find the geometry of the turning motion
-	Vector2f turning_center = BaseLink2Odom({0, turning_radius});
-
-	Vector2f start_point = (turning_direction == "left" ? odom_loc_ : BaseLink2Odom(path.obstruction));
-	Vector2f end_point   = (turning_direction == "left" ? BaseLink2Odom(path.obstruction) : odom_loc_);
-
-	// Iterate through obstacles to find the one with the minimum clearance
-	float min_clearance = vision_range_;
-	Eigen::Vector2f closest_obs = ObstacleList_.begin()->loc;
+	// Warning: These can be negative
+	float radius = 1/path.curvature;
+	float theta = path.free_path_length/radius;
+
+	// Look 3 car lengths ahead
+	float look_ahead_dist = 3*car_length_;
+
+	// Get start and end points
+	Vector2f center(0,radius);
+	Vector2f start_point(0,0);
+	Vector2f end_point(0,0);
+	end_point.x() = radius*sin(theta) + look_ahead_dist*cos(theta);
+	end_point.y() = radius*(1-cos(theta)) + look_ahead_dist*sin(theta);
+
+	// Flip points for isBetween turning direction dependency
+	Vector2f point_1 = (radius > 0 ? start_point : end_point);
+	Vector2f point_2 = (radius > 0 ? end_point : start_point);
+
+	float clearance = 1.0;
+	float min_clearance = 1.0;
+	Vector2f closest_point(0,0);
+	// Get angle between start and end
 	for (const auto &obs : ObstacleList_)
 	{
-		if (isBetween(turning_center, start_point, end_point, obs.loc))
-		{
-			float offset = (obs.loc - turning_center).norm();
-			float pseudo_clearance = abs(offset - abs_radius);
+		Vector2f obs_point = Odom2BaseLink(obs.loc);
 
-			if (pseudo_clearance < min_clearance) 
+		if (isBetween(center, point_1, point_2, obs_point))
+		{
+			float radius_to_point = (center-obs_point).norm();
+			clearance = std::abs(radius_to_point - std::abs(radius));
+			if (clearance < min_clearance)
 			{
-				closest_obs = obs.loc; 
-				min_clearance = pseudo_clearance;
+				min_clearance = clearance;
+				closest_point = obs_point;
 			}
 		}
 	}
-
-	path.closest_point = Odom2BaseLink(closest_obs);
-	path.clearance = std::min(min_clearance, clearance_limit_);
+	path.clearance = min_clearance;
```

Comment:
I want to keep in the clearance_limit_ value. It keeps the clearance from blowing up to really big values when in free space

Reply:
So it is limited to 1m meter (line 365), but this could be done more clearly.